### PR TITLE
Improve mocha-testing lesson (#1241)

### LINKED ISSUE #1241: Improve mocha-testing lesson
https://github.com/Techtonica/curriculum/blob/master/testing-and-tdd/mocha-testing.md

Under **Mocha vs. Jasmine** section, there is a sentence saying "So, why might you choose one **other** the other?". The highlighted word "other" in this sentence could be replaced by "over" which sounds more appropriate.

 I am a GSSoC 20 participant and I would like to work in this issue.

## Additional Changes
- Add a note in the "should" section of Chai that you if you like that style, you can use the  [should library](https://www.npmjs.com/package/should) with mocha, as that is a common reason to use it over Jasmine
- Move this lesson to `/electives` folder
  - update its link in the root README after moving
- Include an async example in Chai and Mocha and remove the incorrect [Misconception No. 2](https://github.com/Techtonica/curriculum/blob/master/testing-and-tdd/mocha-testing.md#common-mistakes--misconceptions)
- Add a challenge: do the Mocha or Chai projects have their own internal tests? Suggest the student read them and see how they are used

### PR DESCRIPTION
**Pull request by Tahanima**
Closes #1241 

- Fixed typo
- Added async example for Mocha
- Added note under `should` section
- Improved indentation
- Removed Misconception no. 2
- Added challenge
- Moved the lesson to `/electives`
- Updated internal links
- Updated `README.md`

Please review the changes and let me know if there is anything I need to add/change.

### CODE REVIEW

File: electives/mocha-testing.md [hunk 1]
```diff
@@ -122,6 +122,28 @@ Mocha String Test
     at Context.<anonymous> (test/myTest.js:5:16)
 ```
 - When calling `assert.equal(a, b)` mocha expects the values to be equivalent, or else the test will fail.
+- If you need to test asynchronous code, Mocha makes it very simple using callbacks. All you need to do is pass a callback function (usually named `done`) to `it` function and invoke the callback function after your test is complete. 
+```javascript
+var assert = require('assert');
```
Comment (alodahl) [inline electives/mocha-testing.md]: Isnt  assert from mocha or chai? Or is the package really called ‘assert’?

File: electives/mocha-testing.md [hunk 2]
```diff
@@ -179,39 +201,37 @@ foo.should.equal('bar');
    - Jasmine is an out-of-the-box library that has test structure, assertions, displays test results, and spies.
    - Mocha is an extensible testing framework that provides a lot of flexibility
    - Chai is an assertion library that can be used with Mocha.
-2. Assertions inside of asynchronous code is ignored, therefore passing. This can lead to a false positive.
```
Comment (alodahl) [inline electives/mocha-testing.md]: This isn’t totally true, but can we replace it with a way to test async code?

File: electives/mocha-testing.md [hunk 3]
```diff
@@ -122,6 +122,28 @@ Mocha String Test
     at Context.<anonymous> (test/myTest.js:5:16)
 ```
 - When calling `assert.equal(a, b)` mocha expects the values to be equivalent, or else the test will fail.
+- If you need to test asynchronous code, Mocha makes it very simple using callbacks. All you need to do is pass a callback function (usually named `done`) to `it` function and invoke the callback function after your test is complete. 
+```javascript
+var assert = require('assert');
+
+function asyncFunction(stringToTest, callback) {
+   setTimeout(function() {
```
Comment (alodahl) [inline electives/mocha-testing.md]: Can you warn them that this is not best practice and why?

File: electives/mocha-testing.md [hunk 4]
```diff
@@ -240,7 +260,7 @@ describe("fizzBuzz", function(){
 
 ### Challenge
 
-No challenge yet!
+Do the Mocha and Chai projects have their own internal tests? Read them and see how they are used.
```
Comment (alodahl) [inline electives/mocha-testing.md]: Is part of the challenge to find the docs or codebase too? If not, please give them some links for where to start.

**Comment by Tahanima** [comment]
Hello @alodahl 
I made the requested changes. Please review the changes and let me know if there is anything I need to add/change.

File: electives/mocha-testing.md (hunk 3 above)
Comment (Tahanima) [reply electives/mocha-testing.md]: Done

File: electives/mocha-testing.md (hunk 4 above)
Comment (Tahanima) [reply electives/mocha-testing.md]: Done

**Comment by Kundan28** [comment]
Updated in scoresheet.

**Pull request merged**